source/client-side-encryption/client-side-encryption.rst

```diff
@@ -708,20 +708,44 @@ ClientEncryption
 
       // Creates a new key document and inserts into the key vault collection.
       // Returns the \_id of the created document as a UUID (BSON binary subtype 4).
-      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): UUID;
 
       // An alias function equivalent to createKey.
-      createDataKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createDataKey(kmsProvider: String, opts: Optional<DataKeyOpts>): UUID;
 
       // Decrypts multiple data keys and (re-)encrypts them with a new masterKey, or with their current masterKey if a new one is not given.
       // Returns a RewrapManyDataKeyResult.
       rewrapManyDataKey(filter: Document, opts: Optional<RewrapManyDataKeyOpts>): RewrapManyDataKeyResult;
 
+      // Removes the key document with the given \_id from the key vault collection.
+      // Returns the result of the internal deleteOne() operation on the key vault collection.
+      deleteKey(id: UUID): DeleteResult;
+
+      // Finds a single key document with the given UUID.
+      // Returns the result of the internal find() operation on the key vault collection.
+      getKey(id: UUID): Iterable<Document>;
+
+      // Finds all documents in the key vault collection.
+      // Returns the result of the internal find() operation on the key vault collection.
+      getKeys(): Iterable<Document>;
+
+      // Adds a keyAltName to the keyAltNames array of the key document in the key vault collection with the given \_id.
+      // Returns the previous version of the key document.
+      addKeyAlternateName(id: UUID, keyAltName: String): Optional<Document>;
+
+      // Removes a keyAltName from the keyAltNames array of the key document in the key vault collection with the given \_id.
+      // Returns the previous version of the key document.
+      removeKeyAlternateName(id: UUID, keyAltName: String): Optional<Document>;
```

Comment:
Any reason these methods use "AlternateName" in the name while "AltName" is used by `getKeyByAltName`? Should they be consistent? I'm sure the common spec rules for naming deviations would permit drivers to use either, but it'd be nice for the spec example itself to be consistent.

Reply:
I don't think driver specs should forsake their own [consistency](https://github.com/mongodb/specifications/blob/master/README.rst#strive-to-be-idiomatic-but-favor-consistency) to comply with inconsistencies in other tools (assuming `mongosh` has different variations).

Drivers will already have the flexibility to use different names here, so this probably only directly affects how spec tests would be written. [CSFLE: User Facing API](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#user-facing-api) only talks about options, but [CRUD: Deviations](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#id1) discusses methods as well.

It's not my intention to bikeshed, so I'm happy to defer to @kevinAlbs or another lead to make the call.

Reply:
I am also in favor of renaming these to `addKeyAltName` and `removeKeyAltName` for consistency in Drivers spec. However, the new `createKey` function was added despite already having `createDataKey` solely for consistency with mongosh, which compelled me to side with consistency with mongosh. I would also like to defer on making this call.

Reply:
I didn't realize `createKey` was added purely for `mongosh`. That said, having an extra alias didn't trigger my spider sense as much as introducing two different naming conventions in the same PR :)

Reply:
Fwiw, if this spec introduces names that deviate from the ones currently used in mongosh, we’ll probably want to add aliases for those in mongosh as well.

Reply:
I think that’s okay, yes. If you do, the ticket would be appreciated :)

Reply:
Filed https://jira.mongodb.org/browse/MONGOSH-1239